### PR TITLE
Generic Pool Update

### LINKED ISSUE #7068: options.pool.validateConnection
Hi,
the option is mentioned here: https://github.com/sequelize/sequelize/blob/master/docs/api/sequelize.md

Searching the repository it seems to be only mentioned in docs: https://github.com/sequelize/sequelize/search?utf8=%E2%9C%93&q=validateConnection

And when I tried to do this with latest sequelize, nothing seems to happen:
```
const poolConfig = {
    min: 1,
    max: maxConnections,
    idle: 60000,
    validateConnection: function (arg1) {
      console.log('validateConnection', this, arg1);
      return false;
    }
  };
```

Is this feature actually implemented?

### PR DESCRIPTION
**Pull request by sushantdhiman**
### Pull Request check-list

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does your issue contain a link to existing issue or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Have you added an entry under `Future` in the changelog?

### Description of change

- Bump to latest version of generic pool
- Fix issue where replication priorities were not used properly

Fix #7068, #4711
Close #5776

### CODE REVIEW

**Comment by janmeier** [comment]
Seems like a reasonable approach @sushantdhiman , ping me when you need i final review

**Comment by sushantdhiman** [comment]
@janmeier Ping

File: lib/dialects/abstract/connection-manager.js [hunk 1]
```diff
@@ -51,47 +56,68 @@ class ConnectionManager {
...
-        name: 'sequelize-connection',
-        create: (callback) => {
-          this._connect(config).nodeify((err, connection) => {
-            debug(`pool created max/min: ${config.pool.max}/${config.pool.min} with no replication`);
-            callback(err, connection); // For some reason this is needed, else generic-pool things err is a connection or some shit
-          });
-        },
+      this.pool = Pooling.createPool({
+        create: () => new Promise((resolve) => {
+          this
```
Comment (janmeier) [inline lib/dialects/abstract/connection-manager.js]: No need to wrap in a new promise, you can just return the chain

File: lib/dialects/abstract/connection-manager.js [hunk 2]
```diff
@@ -118,67 +144,91 @@ class ConnectionManager {
...
+        return Promise.join(
+          this.pool.read.clear(),
+          this.pool.write.clear()
+        );
       },
-      drain: (cb) => {
-        this.pool.write.drain(() => {
-          this.pool.read.drain(cb);
+      drain: () => {
+        return this.pool.write.drain(() => {
```
Comment (janmeier) [inline lib/dialects/abstract/connection-manager.js]: These should be able to drain in parallel

File: lib/dialects/postgres/index.js [hunk 3]
```diff
@@ -12,8 +12,6 @@ class PostgresDialect extends AbstractDialect {
     super();
     this.sequelize = sequelize;
     this.connectionManager = new ConnectionManager(this, sequelize);
-    this.connectionManager.initPools();
```
Comment (janmeier) [inline lib/dialects/postgres/index.js]: already handled in the call to `super`, right?

Comment (sushantdhiman) [inline lib/dialects/postgres/index.js]: Yes

File: lib/dialects/abstract/connection-manager.js [hunk 4]
```diff
@@ -6,40 +6,45 @@ const _ = require('lodash');
...
-      config.pool =_.defaults(config.pool || {}, defaultPoolingConfig, {
-        validate: this._validate.bind(this)
-      }) ;
-    } else {
+    if (config.pool === false) {
       throw new Error('Support for pool:false was removed in v4.0');
     }
 
+    config.pool =_.defaults(config.pool || {}, defaultPoolingConfig, {
+      validate: this._validate.bind(this)
```
Comment (sushantdhiman) [inline lib/dialects/abstract/connection-manager.js]: Good idea, user will be able to override it as well

File: lib/dialects/abstract/connection-manager.js (hunk 1 above)
Comment (sushantdhiman) [inline lib/dialects/abstract/connection-manager.js]: We dont want to resolve if there is an error

**Pull request merged**